File: docs/src/main/scala/sexamples/simulation/pingpongglobal/Ponger.scala
package sexamples.simulation.pingpongglobal

import se.sics.kompics.sl._
import se.sics.kompics.network.Network

class Ponger extends ComponentDefinition {

  val self = cfg.getValue[TAddress]("pingpong.ponger.addr");

  val net = requires[Network];

  private var counter: Long = 0L;

  net uponEvent {
    case TMessage(header, Ping) => {
      counter += 1L;
      log.info(s"Got Ping #${counter}!");
      trigger(TMessage(self, header.getSource(), Pong) -> net);
    }
  }
}

File: docs/src/main/java/jexamples/simulation/pingpong/Ponger.java
package jexamples.simulation.pingpong;

import se.sics.kompics.ClassMatchedHandler;
import se.sics.kompics.ComponentDefinition;
import se.sics.kompics.Positive;
import se.sics.kompics.Handler;
import se.sics.kompics.network.Network;
import se.sics.kompics.network.Transport;

public class Ponger extends ComponentDefinition {

  Positive<Network> net = requires(Network.class);

  private long counter = 0;
  private final TAddress self;

  public Ponger() {
    this.self = config().getValue("pingpong.ponger.addr", TAddress.class);
  }

  ClassMatchedHandler<Ping, TMessage> pingHandler =
      new ClassMatchedHandler<Ping, TMessage>() {
        @Override
        public void handle(Ping content, TMessage context) {
          counter++;
          logger.info("Got Ping #{}!", counter);
          trigger(new TMessage(self, context.getSource(), Transport.TCP, Pong.EVENT), net);
        }
      };

  {
    subscribe(pingHandler, net);
  }
}
